### ISSUE TITLE
[1.12] Matter Fabricator energy/scrap consumption

### ISSUE THREAD

**Issue by Skillotic0703** [TechReborn/TechReborn#1226]
The Matter Fabricator uses too little energy and too much scrap, it's completely OP. In GregTech, it used to consume 8192 EU/t, 16, 666, 666 EU per UU-Matter and 33.3 scrap per UU-Matter but it used to use a percentage system so it can not consume this quantity of energy/scrap with the Matter Fabricator from TechReborn.
So, my suggestion is to consume 8192 EU/t and during 2048 ticks which means that the Matter Fabricator will consume 16, 777, 216 EU. And about scrap consumption, it should consume 1 scrap every 64 ticks so every 524 288 EU. So it consumes 32 scrap per UU-Matter. And the problem is that it should consume 27 time less scrapbox. So the best solution would just be to not accept scrapboxes or just consume 1 scrapbox per UU-Matter. But please do something because in its current state, the Matter Fabricator is completely OP.

**Comment by Vectrobe** [comment]
The scrap use is about right IMO, but each UUM needs only 1,600FE, obviously 4 zeros are missing somewhere...

**Comment by Vectrobe** [comment]
Looking at the source, the energy per amp value isn't calculated correctly, in fact it isn't calculated at all...

For each 1 value of AMP you need to pull 1666 FE, so at line 122 of TileMatterFabricator you need to add something like;
int feAmpFct = amp * 1666;
and replace the 85 with said value;

```
int feAmpFct = amp * 1666;
if (amp != 0 && this.canUseEnergy(feAmpFct )) {
this.useEnergy(feAmpFct );
...
```

**Comment by Paul17041993** [comment]
Tell a lie, FE is 4 times the value of IE, and RF 4 times the value over that, so the factor would be 26,656...

**Comment by Vectrobe** [comment]
RF is currently broken as well however, skewing the numbers...
#1230

**Comment by Vectrobe** [comment]
#1231

**Comment by drcrazy** [comment]
Fixed with #1231
